File: downloadlib/src/main/java/com/king/mobile/downloadlib/model/ThreadTask.java
package com.king.mobile.downloadlib.model;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "download_thread_task")
public class ThreadTask {
    @Ignore
    private static final long serialVersionUID = 8822818790694831649L;
    @PrimaryKey(autoGenerate = true)
    public long id;
    @ColumnInfo(defaultValue = "0")
    public String url;
    @ColumnInfo(defaultValue = "0")
    public long size;
    @ColumnInfo(defaultValue = "0")
    public long start;
    @ColumnInfo(defaultValue = "0")
    public long finished;
    @ColumnInfo(defaultValue = "0")
    public boolean isFinished;
    @ColumnInfo
    public long createAt;

    @Override
    public String toString() {
        return "ThreadTask{" +
                "id=" + id +
                ", url='" + url + '\'' +
                ", size=" + size +
                ", start=" + start +
                ", finished=" + finished +
                ", isFinished=" + isFinished +
                ", createAt=" + createAt +
                '}';
    }
}

File: downloadlib/src/main/java/com/king/mobile/downloadlib/ThreadTask.kt
package com.king.mobile.downloadlib

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * 任务下载中的进度和状态
 */
@Entity(tableName = "TAB_THREAD_TASK")
data class ThreadTask(
        @PrimaryKey val id: Long,
        @ColumnInfo val url: String, // 地址
        @ColumnInfo val size: Long, // 任务段的长度
        @ColumnInfo(defaultValue = "0") var start: Long, // 任务的开始位置
        @ColumnInfo(defaultValue = "0") var finished: Long, // 已完成的进度
        @ColumnInfo(defaultValue = "false") var isFinished: Boolean, // 是否完成
        @ColumnInfo val createdAt: Long // 创建时间
)
